internal/mode/static/metrics/collector.go

```diff
@@ -0,0 +1,115 @@
+package metrics
+
+import (
+	"time"
+
+	"github.com/prometheus/client_golang/prometheus"
+)
+
+// ManagerMetricsCollector implements ManagerCollector interface and prometheus.Collector interface
+type ManagerMetricsCollector struct {
+	// Metrics
+	reloadsTotal    prometheus.Counter
+	reloadsError    prometheus.Counter
+	configStale     prometheus.Gauge
+	reloadsDuration prometheus.Histogram
+}
+
+// NewManagerMetricsCollector creates a new ManagerMetricsCollector
+func NewManagerMetricsCollector(constLabels map[string]string) *ManagerMetricsCollector {
+	nc := &ManagerMetricsCollector{
+		reloadsTotal: prometheus.NewCounter(
+			prometheus.CounterOpts{
+				Name:        "nginx_reloads_total",
+				Namespace:   metricsNamespace,
+				Help:        "Number of successful NGINX reloads",
+				ConstLabels: constLabels,
+			}),
+		reloadsError: prometheus.NewCounter(
+			prometheus.CounterOpts{
+				Name:        "nginx_reload_errors_total",
+				Namespace:   metricsNamespace,
+				Help:        "Number of unsuccessful NGINX reloads",
+				ConstLabels: constLabels,
+			},
+		),
+		configStale: prometheus.NewGauge(
+			prometheus.GaugeOpts{
+				Name:        "nginx_stale_config",
+				Namespace:   metricsNamespace,
+				Help:        "Indicates if NGINX is not serving the latest configuration.",
+				ConstLabels: constLabels,
+			},
+		),
+		reloadsDuration: prometheus.NewHistogram(
+			prometheus.HistogramOpts{
+				Name:        "nginx_reloads_milliseconds",
+				Namespace:   metricsNamespace,
+				Help:        "Duration in milliseconds of NGINX reloads",
+				ConstLabels: constLabels,
+				Buckets:     []float64{100.0, 200.0, 300.0, 400.0, 500.0},
```

Comment:
I suggest using buckets with bigger values
0.5s, 1s, 5s,  10s, 30s

Based on my experience with NIC, we always had at least 200ms 
After that, the reload time grows with the number of servers + TLS certs per server. 
Additionally, reload will depend CPU resource available.

Our current timeout (internal/mode/static/nginx/runtime/manager.go:22) for reload is 60s -- so I think it makes sense to have a bucket with a value close to it.


We can also try to get more realistic buckets by doing performance testing and trying to figure out normal reload times.

Reply:
Yeah, I figured we'd get better bucket sizes when we do some performance testing. I've changed them to the values you suggested, although I kept it as milliseconds for granularity in the `sum` histogram function.